snapcraft/application.py

```diff
@@ -163,11 +182,35 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
                 yaml_data = util.safe_yaml_load(file)
             base = yaml_data.get("base")
             build_base = yaml_data.get("build-base")
+            _get_esm_error_for_base(base)
             if "core24" in (base, build_base) or build_base == "devel":
                 # We know for sure that we're handling a core24 project
                 self._known_core24 = True
             elif any(arg in ("version", "--version", "-V") for arg in sys.argv):
                 pass
+            elif "remote-build" in sys.argv and any(
```

Comment:
Moving this out of the command means there's a mismatch now - this code in `snapcraft.application` is being tested in `tests.unit.commands.test_remote`.

Could you move either the code or the tests?

Reply:
OK, I moved the tests